### ISSUE TITLE
Looking for help solving a problem @import solves, but @use doesn't

### ISSUE THREAD

**Issue by FlipLucky** [sass/sass#3771]
Hello, 

I am posting this question, because I have tried several options, but I keep getting stuck.

> In summary, I am building a layered SCSS structure with base variables, that need to be overwritten. I am trying to avoid `@import`, since this will be phased out. A colleague of mine "solved" my problem with `@import`, and I cannot get it to work with` @use`.

The structure I am building is for a CMS, with multiple instances of the same codebase. I am building this with the `@layer`   from css, to mitigate specificity issues. 

The structure is as followed: 

- base layer -> should include base components, default colors and fonts
- structure layer -> handling layouts of base components
- theme -> styling variations split into different themes, overwriting structure + base
- customer -> this layer should overwrite the base colorscheme + fonts -> this is not available in dev, only on production
- custom -> specific content related css which could adjust all of the above

The behaviour that is needed is as follows: 
base components should implement default colors and fonts to create a rigid structure, that will always have a fallback, in case the production website does not have the required colors or font settings.

However, the customer layer should be able to overwrite the default colors and fonts, and thus creating a dynamic style system which will update to the customers needs. 

`@import`   will import the base files, and after that the customer styles, and since `@import`  makes the variables global

_this will be a tough one to keep it short..._

The way I set up the system is: base with an index file and colors
I declare use on base ( which will forward the colors ), so I can access the color with for example`  base.$primary-color` 
My idea was to overwrite base with a `@use .. with`
This triggers an error which states that base colors are already being used, and cannot be overwritten. 

tldr; Is there a way I can create a solution with use, without import and thus not creating global variables?

Thank you for your patience, and effort to read this wall of text.

**Comment by FlipLucky** [comment]
I have found a solution after a LOT of tryouts: 
I needed to forward the base colors in the customer colors to make it work: 
_base.scss:

```
$primary-color: grey !default;
$secondary-color: lightgrey !default;
```

_customer.scss:
```
@forward './base-color-scheme';

$primary-color: red;
```
_index.scss:

```
@forward 'colors/base-color-scheme';
@forward 'colors/customer-color-scheme';
```
If customer is empty, base will still work
if customer forwards, but does not forward every variable, it will still fall back to the originals.
